infra/examples/gcp-hris/main.tf

```diff
@@ -0,0 +1,148 @@
+terraform {
+  required_providers {
+    google = {
+      version = "4.12.0"
+    }
+  }
+
+  # if you leave this as local, you should backup/commit your TF state files
+  backend "local" {
+  }
+}
+
+# NOTE: if you don't have perms to provision a GCP project in your billing account, you can have
+# someone else create one and than import it:
+#  `terraform import google_project.psoxy-project your-psoxy-project-id`
+# either way, we recommend the project be used exclusively to host psoxy instances corresponding to
+# a single worklytics account
+resource "google_project" "psoxy-project" {
+  name            = "Psoxy - ${var.environment_name}"
+  project_id      = var.project_id
+  folder_id       = var.folder_id
+  billing_account = var.billing_account_id
+}
+
+module "psoxy-gcp" {
+  source = "../../modules/gcp"
+
+  project_id          = google_project.psoxy-project.project_id
+  invoker_sa_emails   = var.worklytics_sa_emails
+
+  depends_on = [
+    google_project.psoxy-project
+  ]
+}
+
+module "psoxy-package" {
+  source = "../../modules/psoxy-package"
+
+  implementation     = "gcp"
+  path_to_psoxy_java = "../../../java"
+}
+
+data "archive_file" "source" {
+  type        = "zip"
+  source_file  = module.psoxy-package.path_to_deployment_jar
+  output_path = "/tmp/function.zip"
+}
+
+# Create bucket that will host the source code
+resource "google_storage_bucket" "deployment_bucket" {
+  name            = "${var.bucket_prefix}-function"
+  location        = var.bucket_location
+  force_destroy   = true
+  project         = google_project.psoxy-project.project_id
+}
+
+# Add source code zip to bucket
+resource "google_storage_bucket_object" "function" {
+  # Append file MD5 to force bucket to be recreated
+  name   = format("${module.psoxy-package.filename}#%s", formatdate("mmss", timestamp()))
+  content_type = "application/zip"
+  bucket = google_storage_bucket.deployment_bucket.name
```

Comment:
This is for deploying the function. The JAR needs to be zipped inside a file; and that JAR should be the unique file present in that ZIP. Then, with ` source_archive_bucket`  and `source_archive_object` properties in cloud function they could refer to a location in the storage. Depending on the format of the file (ZIP with many files = source code which will be compiled during cloud function deployment; ZIP with single file = binary to deploy) it will deploy the cloud function with the code included.